src/cloudflare/internal/images-api.ts

```diff
@@ -40,8 +40,16 @@ class TransformationResultImpl implements ImageTransformationResult {
     return contentType;
   }
 
-  public image(): ReadableStream<Uint8Array> {
-    return this.bindingsResponse.body || new ReadableStream();
+  public image(
+    options?: ImageTransformationOutputOptions
+  ): ReadableStream<Uint8Array> {
+    let stream = this.bindingsResponse.body || new ReadableStream();
+
+    if (options?.encoding === 'base64') {
+      stream = stream.pipeThrough(createBase64EncoderTransformStream());
+    }
+
+    return stream;
```

Comment:
Right now, we always create a new ReadableStream even though we override it in the next statement. This would avoid creating unnecessary readable streams.

```suggestion
    return options?.encoding === 'base64' ?
      stream.pipeThrough(createBase64EncoderTransformStream()) : (this.bindingsResponse.body || new ReadableStream());
```

Reply:
`||` shortcircuits so we'll only create the empty stream if it is needed here (also changed to use `Blob` anyway due to comment further down)